src/codegen/utilities/interpreter_helpers.py

```diff
@@ -394,14 +402,17 @@ def get_response_parameters(func):
     output_parameters = get_output_params(func)
     for parameter in output_parameters:
         if not parameter.repeating_argument:
+            name = parameter.parameter_name
+            if parameter.parameter_name in MODIFIED_INTERPRETER_PARAMS:
+                name = MODIFIED_INTERPRETER_PARAMS.get(parameter.parameter_name)
             if is_read_method and "read_array" in parameter.parameter_name:
```

Comment:
Did you intend to replace the other references to `param.parameter_name` with `name`?

Reply:
No Brad, the expectation is to assign `r0 = r_0`, where `param.parameter_name` is `r_0` and the modified name is `r0`